File: Backend/src/main/java/com/testing/ai_api_testing_platform/service/ApiRequestService.java
package com.testing.ai_api_testing_platform.service;

import com.testing.ai_api_testing_platform.domain.entity.ApiRequest;
import com.testing.ai_api_testing_platform.domain.entity.ApiResponse;
import com.testing.ai_api_testing_platform.domain.entity.User;
import com.testing.ai_api_testing_platform.dto.request.ApiRequestCreateUpdateRequest;
import com.testing.ai_api_testing_platform.dto.response.ApiRequestExecuteResponse;
import com.testing.ai_api_testing_platform.dto.response.ApiRequestResponse;
import com.testing.ai_api_testing_platform.dto.response.ApiResponseResponse;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
@Service
public interface ApiRequestService {
    public ApiRequestResponse create(String email, ApiRequestCreateUpdateRequest request) ;
    public List<ApiRequestResponse> listMine(String email) ;
    public ApiRequestResponse getMine(String email, Long requestId);
    public ApiRequestResponse updateMine(String email, Long requestId, ApiRequestCreateUpdateRequest request) ;
    public void deleteMine(String email, Long requestId);
    public ApiRequestExecuteResponse executeMine(String email, Long requestId) ;
    public List<ApiResponseResponse> listRecentResponses(String email, Long requestId);
}

File: Frontend/src/services/apiRequestService.js
import apiClient from './apiClient';

export const apiRequestService = {
  async create(payload) {
    const response = await apiClient.post('/api-requests', payload);
    return response.data;
  },

  async listMine() {
    const response = await apiClient.get('/api-requests');
    return Array.isArray(response.data) ? response.data : [];
  },

  async updateMine(requestId, payload) {
    const response = await apiClient.put(`/api-requests/${requestId}`, payload);
    return response.data;
  },

  async deleteMine(requestId) {
    await apiClient.delete(`/api-requests/${requestId}`);
  },

  async executeMine(requestId) {
    const response = await apiClient.post(`/api-requests/${requestId}/execute`);
    return response.data;
  },

  async listRecentResponses(requestId) {
    const response = await apiClient.get(`/api-requests/${requestId}/responses`);
    return Array.isArray(response.data) ? response.data : [];
  },

  async getMine(requestId) {
    const response = await apiClient.get(`/api-requests/${requestId}`);
    return response.data;
  },
};
